src/java/org/apache/cassandra/db/ReadCommand.java

```diff
@@ -632,6 +654,84 @@ private void maybeDelayForTesting()
         }
     }
 
+    private UnfilteredPartitionIterator withQuerySizeTracking(UnfilteredPartitionIterator iterator)
+    {
+        if (!trackWarnings || SchemaConstants.isSystemKeyspace(metadata().keyspace)) // exclude internal keyspaces
+            return iterator;
+        final long warnThresholdBytes = DatabaseDescriptor.getLocalReadTooLargeWarningThresholdKb() * 1024;
+        final long abortThresholdBytes = DatabaseDescriptor.getLocalReadTooLargeAbortThresholdKb() * 1024;
+        if (warnThresholdBytes == 0 && abortThresholdBytes == 0)
+            return iterator;
+        class QuerySizeTracking extends Transformation<UnfilteredRowIterator>
+        {
+            private DecoratedKey currentKey;
```

Comment:
nit: I think `currentKey` can just be a local in `applyToPartition()`

Reply:
original patch I tried being fancy then released this is the straw which broke the camels back; may not be the problematic key...  so removed... will clean up